src/settings/ServicePoints/ServicePointFormContainer.js

```diff
@@ -3,21 +3,36 @@ import React, {
   useEffect,
   useState,
 } from 'react';
+import PropTypes from 'prop-types';
 import {
   cloneDeep,
   unset,
   orderBy,
+  set,
 } from 'lodash';
 
 import ServicePointForm from './ServicePointForm';
+import {
+  shortTermExpiryPeriod,
+  shortTermClosedDateManagementMenu,
+  longTermClosedDateManagementMenu
+} from './constants';
 
 const ServicePointFormContainer = ({
   onSave,
   parentResources,
   initialValues: servicePoint,
   ...rest
 }) => {
-  const [initialValues, setInitialValues] = useState(servicePoint);
+  const getServicePoint = () => {
+    // remove holdShelfClosedLibraryDateManagement from servicepoint object when pickupLocation is not true
+    if (!servicePoint.pickupLocation) {
+      unset(servicePoint, 'holdShelfClosedLibraryDateManagement');
```

Comment:
`unset` is a mutable method, and here it mutates the parent property - this is bad practice. Please use the `omit` method instead.